### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by phamann**
Had to remove v1 for another refactor i'm doing to support multiple DB connections within the api routes. Ended up going down a bit of a rabbit hole and being much bigger than I was hoping. But worth it.

- Removed v1 server and tests folder
- Fix app init which was coupled to v1 :scream: 
- Fix flaky v2 tests which relied on v1 tests :scream_cat: 
- Refactor flaky tests which assumed empty DB
- Fixed linting errors and warnings

The tests are in a better state, but still have some messy DB output which requires further investigation but out of the scope of this PR. At least they all pass #yolo

### CODE REVIEW

File: jobs/engagement.js [hunk 1]
```diff
@@ -62,8 +62,8 @@ const nowQueryStrings = [
 	RETURN p IS NOT NULL AS opted_into_weekly_email`
 ];
 
-const followingQuery = `MATCH (user:User {uuid: {userId}})-[:followed]-(c:Concept)
-						RETURN COUNT(c) AS following`;
+// const followingQuery = `MATCH (user:User {uuid: {userId}})-[:followed]-(c:Concept)
+// 						RETURN COUNT(c) AS following`;
```
Comment (laurieboyes) [inline jobs/engagement.js]: Left-in debugging?

File: tests/v2/node.spec.js [hunk 2]
```diff
@@ -37,16 +37,12 @@ describe('API for a graph node', function () {
 				});
 		});
 
-		it('POST should not create a node that doesn\'t match validation rules', function (done) {
+		it(`POST should not create a node that doesn't match validation rules`, function (done) {
 
 			request(api)
 				.post('/v2/user')
 				.send({uuid: 'not-a-uuid'})
-				.expect(200)
```
Comment (keirog) [inline tests/v2/node.spec.js]: WAT

**Comment by laurieboyes** [comment]
LGTM apart from comment

File: server/v2/models/index.js [hunk 3]
```diff
@@ -2,11 +2,13 @@
 
 const db = require('../../utils/db');
 const directly = require('directly');
-const models = {};
-
-require("fs").readdirSync(__dirname).forEach(file => {
-  models[file.replace('.js', '')] = require(`./${file}`);
-});
+const models = require("fs")
```
Comment (keirog) [inline server/v2/models/index.js]: Single-quote police

File: jobs/engagement.js (hunk 1 above)
Comment (phamann) [inline jobs/engagement.js]: Ah yes @laurieboyes I wanted to ask you about this. Linting told me it wasn't being used in the file. Therefore the next question is why was it written in the first place? Do we want this information in the engagement business logic or output?

File: tests/v2/node.spec.js (hunk 2 above)
Comment (phamann) [inline tests/v2/node.spec.js]: Haha, yeah don't quite understand how it ever worked! But I did learn that you can pass `done` as a second parameter to `.expect(500, done);`

**Comment by keirog** [comment]
:+1:

File: tests/v2/node.spec.js (hunk 2 above)
Comment (ironsidevsquincy) [inline tests/v2/node.spec.js]: it's async, so the test was ending before this was run

top tip, you can just return promises from tests, and it knows to wait, e.g.

```
return request(api)
    .post('/v2/user')
    .send({uuid: 'not-a-uuid'})
    .expect(500)
```

File: jobs/engagement.js (hunk 1 above)
Comment (phamann) [inline jobs/engagement.js]: Test

Comment (laurieboyes) [inline jobs/engagement.js]: @phamann Oh yeah whoops! That's a line I accidentally left in before. It's in the `nowQueries` array, and these lines can be safely removed

Comment (phamann) [inline jobs/engagement.js]: :+1:

File: tests/v2/node.spec.js (hunk 2 above)
Comment (keirog) [inline tests/v2/node.spec.js]: :sparkles:

**Pull request merged**